File: 卡码/代码随想录/1/06-二叉树/Js/01-二叉树遍历方式/LeetCode_0144_0145_0094.js
//144-145-94.二叉树的前/后/中序遍历
import TreeNode from "../TreeNode.js";

//先序遍历
const preorderTraversal = function(root) {
  let res=new Array();
  preOrder(root,res);
  return res;
};
const preOrder=(root,res)=>{
  if(root==null) return;
  res.push(root.val);
  preOrder(root.left,res);
  preOrder(root.right,res);
};
//中序遍历
const inorderTraversal=function(root){
  const res=new Array();
  inOrder(root,res);
  return res;
};
const inOrder=(root,res)=>{
  if(root==null) return;
  inOrder(root.left,res);
  res.push(root.val);
  inOrder(root.right,res);
};
//后序遍历
const postorderTraversal=function(root){
  const res=new Array();
  postOrder(root,res);
  return res;
};
const postOrder=(root,res)=>{
  if(root==null) return;
  postOrder(root.left,res);
  postOrder(root.right,res);
  res.push(root.val);
};

File: 卡码/代码随想录/1/06-二叉树/Go/01-二叉树遍历方式/LeetCode_0144_0145_0094.go
//144-145-94.二叉树的前/后/中序遍历
package main

import (
	. "binaryTree/types"
)

//先序遍历
func preorderTraversal(root *TreeNode) []int {
	res:=make([]int,0)
	preOrder(root,&res)
	return res
}
func preOrder(root *TreeNode,v *[]int){
	if root==nil {
		return
	}
	*v=append(*v,root.Val)
	preOrder(root.Left,v)
	preOrder(root.Right,v)
}
//中序遍历
func inorderTraversal(root *TreeNode) []int {
	res:=make([]int,0)
	inOrder(root,&res)
	return res
}
func inOrder(root *TreeNode,v *[]int){
	if root==nil {
		return
	}
	inOrder(root.Left,v)
	*v=append(*v,root.Val)
	inOrder(root.Right,v)
}
//后序遍历
func postorderTraversal(root *TreeNode) []int {
	res:=make([]int,0)
	postOrder(root,&res)
	return res
}
func postOrder(root *TreeNode,v *[]int){
	if root==nil {
		return
	}
	postOrder(root.Left,v)
	postOrder(root.Right,v)
	*v=append(*v,root.Val)
}
